### ISSUE TITLE
[Question] Code generation examples

### ISSUE THREAD

**Issue by saona-raimundo** [rust-analyzer/ungrammar#51]
Thanks for the crate!

[the post](https://rust-analyzer.github.io/blog/2020/10/24/introducing-ungrammar.html) mentions that code generation can be used based on the parsed grammar given by ungrammar.

Would you have some examples of such a code generation? 
Maybe something in rust-analyzer? 

Thanks beforehand!

Note: This is my first attempt at doing code-generation, so some high-level guide would also be useful :)

**Comment by Veykril** [comment]
https://github.com/rust-lang/rust-analyzer/blob/master/crates/syntax/src/tests/sourcegen_ast.rs and https://github.com/rome/tools/tree/main/xtask/codegen are two examples

**Comment by matklad** [comment]
Wait WAT? Rome uses ungrammar? 🤯

**Comment by Veykril** [comment]
Yes, well, a fork :) They implemented the leading/trailing trivia model (which I am planning to swap r-a to at some point as it seems to have a lot more benefits)
